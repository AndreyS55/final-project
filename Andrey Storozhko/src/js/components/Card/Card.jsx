import React from 'react';
import CatchButton from '../CatchButton/CatchButton';
import PokemonImage from '../PokemonImage/PokemonImage';

import styles from './Card.scss';

const Card = (props) => {
  const { id, name, catchPokemon, date, catched } = props;
  return (
    <li className={styles.pokemonsItem} onClick={() => props.handleOpen(id)}>
      <figure className={styles.pokemonImageWrapper}>
        <PokemonImage id={id} />
      </figure>
      <div className={styles.cardText}>{name.toUpperCase()}</div>
      <CatchButton name={name} id={id} catchPokemon={catchPokemon} date={date} catched={catched} />
    </li>
  );
};

export default Card;
