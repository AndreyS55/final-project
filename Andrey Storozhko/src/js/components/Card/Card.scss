.pokemonsItem {
  width: 150px;
  padding: 5px;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 1px solid black;
  border-top: 1px solid black;
  border-left: 1px solid black;
  margin-top: -1px;
  margin-left: -1px;
  background-color: white;

  &:hover {
    box-shadow: inset 0 0 80px -20px rgba(0, 0, 0, 0.5);
  }
}

.pokemonImageWrapper {
  max-width: 100px;
}

.cardText {
  padding: 5px 0;
}
