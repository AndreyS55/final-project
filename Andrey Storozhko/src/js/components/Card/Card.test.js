import React from 'react';
import Card from './Card';

it('should render a Card', () => {
  const wrapper = shallow(<Card name="pikachu" />);
  expect(wrapper).toMatchSnapshot();
});
