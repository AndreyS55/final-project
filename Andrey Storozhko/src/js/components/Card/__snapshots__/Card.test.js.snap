// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Card 1`] = `
<li
  className="pokemonsItem"
  onClick={[Function]}
>
  <figure
    className="pokemonImageWrapper"
  >
    <PokemonImage />
  </figure>
  <div
    className="cardText"
  >
    PIKACHU
  </div>
  <CatchButton
    name="pikachu"
  />
</li>
`;
